Roslyn Compilation, 160 errors and process terminated due to NullReferenceException when a class with primary constructor implements INotifyPropertyChanged

**Version Used**: 
C# 12, dotnet SDK version 8.0.100, MSBuild version 17.8.3+195e7f5a3 for .NET,  Visual Studio Professional 2022 Version 17.8.3
Also occurs when using Rider version 2023.3.2

**Steps to Reproduce**:
1- Create a C# file with the following code and try to compile it. 
```csharp
using System.ComponentModel;

internal class MyClass(MyOtherClass otherClass, string key, int value) : INotifyPropertyChanged
{
    public string MyProperty { get; private set; } = GetValue(otherClass.MyProperty, key, value);
    private PropertyChangedEventHandler? _propertyChanged;


    private static string GetValue(string myProperty, string key, int value)
    {
        throw new NotImplementedException();
    }

    public event PropertyChangedEventHandler? PropertyChanged
    {
        add
        {
            lock (this)
            {
                if (_propertyChanged is null)
                {
                    otherClass.SomethingChanged += OnSomethingChanged;
                }
                _propertyChanged += value;

            }
        }
        remove
        {
            lock (this)
            {
                if (_propertyChanged is null)
                {
                    return;
                }
                _propertyChanged -= value;
                if (_propertyChanged is null)
                {
                    otherClass.SomethingChanged -= OnSomethingChanged;
                }
            }
        }
    }

    private void OnSomethingChanged(object? sender, EventArgs e)
    {
        throw new NotImplementedException();
    }
}

internal class MyOtherClass
{
    public event EventHandler? SomethingChanged;
    public string MyProperty { get; set; }
}

```

**Expected Behavior**:
The code should compile without error.

**Actual Behavior**:

- When building from the command line (`dotnet build`), the C# compiler produces 160 errors the crashes with error `C:\Program Files\dotnet\sdk\8.0.100\Roslyn\Microsoft.CSharp.Core.targets(84,5): error : Process terminated. System.NullReferenceException: Object reference not set to an instance of an object. [D:\source\my\ConsoleApp1\ConsoleApp1\ConsoleApp1.csproj]` (This is a partial log of the error which is very big, the entire error log is attached [here](https://github.com/dotnet/roslyn/files/13751763/result.txt)).
- When building in Visual Studio I get `C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets(84,5): error MSB6006: "csc.exe" exited with code -2146232797.`

Moreover, when working with either Visual Studio or Rider, the C# analyzers and generators stop working (apparently because of the Roslyn process crashing). While this may not be apparent in my sample reproduction, it occurs in my actual project, but I kept the reproduction to minimum and excluded irrelevant code. 

This seems to be similar to #50198, however the old issue is resolved (may be this is a regression?) and old one was related to records and not primary constructors.

**Workaround**:

Using a normal constructor instead of a primary constructor with that class fixed the issue and allowed me to continue my work.

--- a/src/Compilers/CSharp/Portable/Symbols/Source/LambdaParameterSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/LambdaParameterSymbol.cs
@@ -10,6 +10,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class LambdaParameterSymbol : SourceComplexParameterSymbolBase
     {
+        private readonly TypeWithAnnotations _parameterType;
         private readonly SyntaxList<AttributeListSyntax> _attributeLists;
 
         public LambdaParameterSymbol(
@@ -24,11 +25,14 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
            bool isDiscard,
            bool isParams,
            Location location)
-           : base(owner, ordinal, parameterType, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis: false, scope)
+           : base(owner, ordinal, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis: false, scope)
         {
+            _parameterType = parameterType;
             _attributeLists = attributeLists;
             IsDiscard = isDiscard;
         }
+
+        public override TypeWithAnnotations TypeWithAnnotations => _parameterType;
 
         public override bool IsDiscard { get; }
 

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs
@@ -39,7 +39,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
         protected SourceComplexParameterSymbolBase(
             Symbol owner,
             int ordinal,
-            TypeWithAnnotations parameterType,
             RefKind refKind,
             string name,
             Location location,
@@ -47,7 +46,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             bool isParams,
             bool isExtensionMethodThis,
             ScopedKind scope)
-            : base(owner, parameterType, ordinal, refKind, scope, name, location)
+            : base(owner, ordinal, refKind, scope, name, location)
         {
             Debug.Assert((syntaxRef == null) || (syntaxRef.GetSyntax().IsKind(SyntaxKind.Parameter)));
 
@@ -393,9 +392,9 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             if (convertedExpression.ConstantValueOpt == null && convertedExpression.Kind == BoundKind.Conversion &&
                 ((BoundConversion)convertedExpression).ConversionKind != ConversionKind.DefaultLiteral)
             {
-                if (parameterType.Type.IsNullableType())
+                if (Type.IsNullableType())
                 {
-                    convertedExpression = binder.GenerateConversionForAssignment(parameterType.Type.GetNullableUnderlyingType(),
+                    convertedExpression = binder.GenerateConversionForAssignment(Type.GetNullableUnderlyingType(),
                         valueBeforeConversion, diagnostics, Binder.ConversionForAssignmentFlags.DefaultParameter);
                 }
             }
@@ -1516,6 +1515,8 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
     internal sealed class SourceComplexParameterSymbol : SourceComplexParameterSymbolBase
     {
+        private readonly TypeWithAnnotations _parameterType;
+
         internal SourceComplexParameterSymbol(
             Symbol owner,
             int ordinal,
@@ -1527,16 +1528,19 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             bool isParams,
             bool isExtensionMethodThis,
             ScopedKind scope)
-            : base(owner, ordinal, parameterType, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis, scope)
+            : base(owner, ordinal, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis, scope)
         {
+            _parameterType = parameterType;
         }
 
+        public override TypeWithAnnotations TypeWithAnnotations => _parameterType;
         public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
     }
 
     internal sealed class SourceComplexParameterSymbolWithCustomModifiersPrecedingRef : SourceComplexParameterSymbolBase
     {
         private readonly ImmutableArray<CustomModifier> _refCustomModifiers;
+        private readonly TypeWithAnnotations _parameterType;
 
         internal SourceComplexParameterSymbolWithCustomModifiersPrecedingRef(
             Symbol owner,
@@ -1550,15 +1554,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             bool isParams,
             bool isExtensionMethodThis,
             ScopedKind scope)
-            : base(owner, ordinal, parameterType, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis, scope)
+            : base(owner, ordinal, refKind, name, location, syntaxRef, isParams, isExtensionMethodThis, scope)
         {
             Debug.Assert(!refCustomModifiers.IsEmpty);
 
+            _parameterType = parameterType;
             _refCustomModifiers = refCustomModifiers;
 
             Debug.Assert(refKind != RefKind.None || _refCustomModifiers.IsEmpty);
         }
 
+        public override TypeWithAnnotations TypeWithAnnotations => _parameterType;
         public override ImmutableArray<CustomModifier> RefCustomModifiers => _refCustomModifiers;
     }
 }

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceEventAccessorSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceEventAccessorSymbol.cs
@@ -18,7 +18,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
         private readonly string _name;
         private readonly ImmutableArray<MethodSymbol> _explicitInterfaceImplementations;
 
-        private ImmutableArray<ParameterSymbol> _lazyParameters;
+        private readonly ImmutableArray<ParameterSymbol> _parameters;
         private TypeWithAnnotations _lazyReturnType;
 
         public SourceEventAccessorSymbol(
@@ -46,6 +46,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                                                 hasThisInitializer: false)))
         {
             _event = @event;
+            _parameters = ImmutableArray.Create<ParameterSymbol>(new SynthesizedEventAccessorValueParameterSymbol(this, 0));
 
             string name;
             ImmutableArray<MethodSymbol> explicitInterfaceImplementations;
@@ -97,8 +98,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
         protected sealed override void MethodChecks(BindingDiagnosticBag diagnostics)
         {
-            Debug.Assert(_lazyParameters.IsDefault != _lazyReturnType.HasType);
-
             // CONSIDER: currently, we're copying the custom modifiers of the event overridden
             // by this method's associated event (by using the associated event's type, which is
             // copied from the overridden event).  It would be more correct to copy them from
@@ -122,9 +121,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
                         _lazyReturnType = TypeWithAnnotations.Create(eventTokenType);
                         this.SetReturnsVoid(returnsVoid: false);
-
-                        var parameter = new SynthesizedAccessorValueParameterSymbol(this, _event.TypeWithAnnotations, 0);
-                        _lazyParameters = ImmutableArray.Create<ParameterSymbol>(parameter);
                     }
                     else
                     {
@@ -136,9 +132,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                         Binder.ReportUseSite(voidType, diagnostics, this.Location);
                         _lazyReturnType = TypeWithAnnotations.Create(voidType);
                         this.SetReturnsVoid(returnsVoid: true);
-
-                        var parameter = new SynthesizedAccessorValueParameterSymbol(this, TypeWithAnnotations.Create(eventTokenType), 0);
-                        _lazyParameters = ImmutableArray.Create<ParameterSymbol>(parameter);
                     }
                 }
                 else
@@ -150,9 +143,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                     Binder.ReportUseSite(voidType, diagnostics, this.Location);
                     _lazyReturnType = TypeWithAnnotations.Create(voidType);
                     this.SetReturnsVoid(returnsVoid: true);
-
-                    var parameter = new SynthesizedAccessorValueParameterSymbol(this, _event.TypeWithAnnotations, 0);
-                    _lazyParameters = ImmutableArray.Create<ParameterSymbol>(parameter);
                 }
             }
         }
@@ -189,9 +179,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
         {
             get
             {
-                LazyMethodChecks();
-                Debug.Assert(!_lazyParameters.IsDefault);
-                return _lazyParameters;
+                return _parameters;
             }
         }
 

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs
@@ -20,7 +20,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
     {
         protected SymbolCompletionState state;
-        protected readonly TypeWithAnnotations parameterType;
         private readonly string _name;
         private readonly Location? _location;
         private readonly RefKind _refKind;
@@ -98,7 +97,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
         protected SourceParameterSymbol(
             Symbol owner,
-            TypeWithAnnotations parameterType,
             int ordinal,
             RefKind refKind,
             ScopedKind scope,
@@ -107,7 +105,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             : base(owner, ordinal)
         {
             Debug.Assert((owner.Kind == SymbolKind.Method) || (owner.Kind == SymbolKind.Property));
-            this.parameterType = parameterType;
             _refKind = refKind;
             _scope = scope;
             _name = name;
@@ -258,14 +255,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                 return IsImplicitlyDeclared ?
                     ImmutableArray<SyntaxReference>.Empty :
                     GetDeclaringSyntaxReferenceHelper<ParameterSyntax>(this.Locations);
-            }
-        }
-
-        public sealed override TypeWithAnnotations TypeWithAnnotations
-        {
-            get
-            {
-                return this.parameterType;
             }
         }
 

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs
@@ -760,7 +760,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
             if (!isGetMethod)
             {
-                parameters.Add(new SynthesizedAccessorValueParameterSymbol(this, _property.TypeWithAnnotations, parameters.Count));
+                parameters.Add(new SynthesizedPropertyAccessorValueParameterSymbol(this, parameters.Count));
             }
 
             return parameters.ToImmutableAndFree();

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceSimpleParameterSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceSimpleParameterSymbol.cs
@@ -15,6 +15,8 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// </summary>
     internal sealed class SourceSimpleParameterSymbol : SourceParameterSymbol
     {
+        private readonly TypeWithAnnotations _parameterType;
+
         public SourceSimpleParameterSymbol(
             Symbol owner,
             TypeWithAnnotations parameterType,
@@ -36,9 +38,12 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             ScopedKind scope,
             string name,
             Location? location)
-            : base(owner, parameterType, ordinal, refKind, scope, name, location)
+            : base(owner, ordinal, refKind, scope, name, location)
         {
+            _parameterType = parameterType;
         }
+
+        public override TypeWithAnnotations TypeWithAnnotations => _parameterType;
 
         public override bool IsDiscard => false;
 

--- a/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAccessorValueParameterSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAccessorValueParameterSymbol.cs
@@ -17,10 +17,10 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// This parameter has no source location/syntax, but may have attributes.
     /// Attributes with 'param' target specifier on the accessor must be applied to the this parameter.
     /// </summary>
-    internal sealed class SynthesizedAccessorValueParameterSymbol : SourceComplexParameterSymbolBase
+    internal abstract class SynthesizedAccessorValueParameterSymbol : SourceComplexParameterSymbolBase
     {
-        public SynthesizedAccessorValueParameterSymbol(SourceMemberMethodSymbol accessor, TypeWithAnnotations paramType, int ordinal)
-            : base(accessor, ordinal, paramType, RefKind.None, ParameterSymbol.ValueParameterName, accessor.TryGetFirstLocation(),
+        public SynthesizedAccessorValueParameterSymbol(SourceMemberMethodSymbol accessor, int ordinal)
+            : base(accessor, ordinal, RefKind.None, ParameterSymbol.ValueParameterName, accessor.TryGetFirstLocation(),
                    syntaxRef: null,
                    isParams: false,
                    isExtensionMethodThis: false,
@@ -91,6 +91,55 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
                 {
                     AddSynthesizedAttribute(ref attributes, SynthesizedAttributeData.Create(property.AllowNullAttributeIfExists));
                 }
+            }
+        }
+    }
+
+    internal sealed class SynthesizedPropertyAccessorValueParameterSymbol : SynthesizedAccessorValueParameterSymbol
+    {
+        public SynthesizedPropertyAccessorValueParameterSymbol(SourcePropertyAccessorSymbol accessor, int ordinal)
+            : base(accessor, ordinal)
+        {
+            Debug.Assert(accessor.Locations.Length <= 1);
+        }
+
+        public override TypeWithAnnotations TypeWithAnnotations => ((PropertySymbol)((SourcePropertyAccessorSymbol)ContainingSymbol).AssociatedSymbol).TypeWithAnnotations;
+    }
+
+    internal sealed class SynthesizedEventAccessorValueParameterSymbol : SynthesizedAccessorValueParameterSymbol
+    {
+        private SingleInitNullable<TypeWithAnnotations> _lazyParameterType;
+
+        public SynthesizedEventAccessorValueParameterSymbol(SourceEventAccessorSymbol accessor, int ordinal)
+            : base(accessor, ordinal)
+        {
+            Debug.Assert(accessor.Locations.Length <= 1);
+        }
+
+        public override TypeWithAnnotations TypeWithAnnotations
+        {
+            get
+            {
+                return _lazyParameterType.Initialize(static (SourceEventAccessorSymbol accessor) =>
+                                                     {
+                                                         SourceEventSymbol @event = accessor.AssociatedEvent;
+
+                                                         if (accessor.MethodKind == MethodKind.EventAdd)
+                                                         {
+                                                             return @event.TypeWithAnnotations;
+                                                         }
+                                                         else if (@event.IsWindowsRuntimeEvent)
+                                                         {
+                                                             TypeSymbol eventTokenType = @event.DeclaringCompilation.GetWellKnownType(WellKnownType.System_Runtime_InteropServices_WindowsRuntime_EventRegistrationToken);
+                                                             // Use-site info is collected by SourceEventAccessorSymbol.MethodChecks
+                                                             return TypeWithAnnotations.Create(eventTokenType);
+                                                         }
+                                                         else
+                                                         {
+                                                             return @event.TypeWithAnnotations;
+                                                         }
+                                                     },
+                                                     (SourceEventAccessorSymbol)this.ContainingSymbol);
             }
         }
     }

--- a/src/Compilers/CSharp/Test/Emit2/Semantics/PrimaryConstructorTests.cs
+++ b/src/Compilers/CSharp/Test/Emit2/Semantics/PrimaryConstructorTests.cs
@@ -19354,6 +19354,293 @@ class C(int y)
         }
 
         [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71400")]
+        public void ParameterCapturing_174_InInterfaceImplementation()
+        {
+            var source =
+@"
+using System;
+
+interface I
+{
+    event EventHandler E;
+}
+
+class C1(int i) : I
+{
+    public event EventHandler E
+    {
+        add { Console.WriteLine(""C1"" + i++); }
+        remove { }
+    }
+}
+
+class C2(int i) : I
+{
+    public event EventHandler E
+    {
+        add { }
+        remove { Console.WriteLine(""C2"" + i++); }
+    }
+}
+
+class C3(int i) : I
+{
+    event EventHandler I.E
+    {
+        add { Console.WriteLine(""C3"" + i++); }
+        remove { }
+    }
+}
+
+class C4(int i) : I
+{
+    event EventHandler I.E
+    {
+        add { }
+        remove { Console.WriteLine(""C4"" + i++); }
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        I c1 = new C1(123);
+        c1.E += null;
+        c1.E += null;
+
+        I c2 = new C2(123);
+        c2.E -= null;
+        c2.E -= null;
+
+        I c3 = new C3(123);
+        c3.E += null;
+        c3.E += null;
+
+        I c4 = new C4(123);
+        c4.E -= null;
+        c4.E -= null;
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, expectedOutput:
+@"C1123
+C1124
+C2123
+C2124
+C3123
+C3124
+C4123
+C4124").VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71400")]
+        public void ParameterCapturing_175_InInterfaceImplementation()
+        {
+            var source =
+@"
+using System;
+
+interface I
+{
+    int P { get; set; }
+}
+
+class C1(int i) : I
+{
+    public int P
+    {
+        get { Console.WriteLine(""C1"" + i++); return 0; }
+        set { }
+    }
+}
+
+class C2(int i) : I
+{
+    public int P
+    {
+        get {  return 0; }
+        set { Console.WriteLine(""C2"" + i++); }
+    }
+}
+
+class C3(int i) : I
+{
+    int I.P
+    {
+        get { Console.WriteLine(""C3"" + i++);  return 0; }
+        set { }
+    }
+}
+
+class C4(int i) : I
+{
+    int I.P
+    {
+        get {  return 0; }
+        set { Console.WriteLine(""C4"" + i++); }
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        I c1 = new C1(123);
+        _ = c1.P;
+        _ = c1.P;
+
+        I c2 = new C2(123);
+        c2.P = 0;
+        c2.P = 0;
+
+        I c3 = new C3(123);
+        _ = c3.P;
+        _ = c3.P;
+
+        I c4 = new C4(123);
+        c4.P = 0;
+        c4.P = 0;
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, expectedOutput:
+@"C1123
+C1124
+C2123
+C2124
+C3123
+C3124
+C4123
+C4124").VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71400")]
+        public void ParameterCapturing_176_InInterfaceImplementation()
+        {
+            var source =
+@"
+using System;
+
+interface I
+{
+    void M();
+}
+
+class C1(int i) : I
+{
+    public void M() { Console.WriteLine(""C1"" + i++); }
+}
+
+class C3(int i) : I
+{
+    void I.M() { Console.WriteLine(""C3"" + i++); }
+}
+
+class Program
+{
+    static void Main()
+    {
+        I c1 = new C1(123);
+        c1.M();
+        c1.M();
+
+        I c3 = new C3(123);
+        c3.M();
+        c3.M();
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, expectedOutput:
+@"C1123
+C1124
+C3123
+C3124").VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/71400")]
+        public void ParameterCapturing_177_InInterfaceImplementation()
+        {
+            var source =
+@"
+#nullable enable
+
+using System;
+using System.ComponentModel;
+
+internal class MyClass(MyOtherClass otherClass, string key, int value) : INotifyPropertyChanged
+{
+    public string MyProperty { get; private set; } = GetValue(otherClass.MyProperty, key, value);
+    private PropertyChangedEventHandler? _propertyChanged;
+
+
+    private static string GetValue(string myProperty, string key, int value)
+    {
+        throw new NotImplementedException();
+    }
+
+    public event PropertyChangedEventHandler? PropertyChanged
+    {
+        add
+        {
+            lock (this)
+            {
+                if (_propertyChanged is null)
+                {
+                    otherClass.SomethingChanged += OnSomethingChanged;
+                }
+                _propertyChanged += value;
+
+            }
+        }
+        remove
+        {
+            lock (this)
+            {
+                if (_propertyChanged is null)
+                {
+                    return;
+                }
+                _propertyChanged -= value;
+                if (_propertyChanged is null)
+                {
+                    otherClass.SomethingChanged -= OnSomethingChanged;
+                }
+            }
+        }
+    }
+
+    private void OnSomethingChanged(object? sender, EventArgs e)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+internal class MyOtherClass
+{
+    public event EventHandler? SomethingChanged;
+    public string MyProperty { get; set; }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+            CompileAndVerify(comp).VerifyDiagnostics(
+                // (57,32): warning CS0067: The event 'MyOtherClass.SomethingChanged' is never used
+                //     public event EventHandler? SomethingChanged;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "SomethingChanged").WithArguments("MyOtherClass.SomethingChanged").WithLocation(57, 32),
+                // (58,19): warning CS8618: Non-nullable property 'MyProperty' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public string MyProperty { get; set; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "MyProperty").WithArguments("property", "MyProperty").WithLocation(58, 19)
+                );
+        }
+
+        [Fact]
         public void CycleDueToIndexerNameAttribute_01()
         {
             var source = @"

